### DISCUSSION TITLE
Anyone having issue detecting devices (Saumsung S22) recently?

### DISCUSSION THREAD

**Discussion by kalto** [mKeRix/room-assistant#1384] [Q&A]
Hi,

I've been using Room-Assistant with Home Assistant for a couple of years. It runs on 4 Pi Zero W spread across the house. Use it to detect my phone (Samsung) and my wife's phone (iPhone). Works mostly well. 

Had some issue recently where I realized that my Room-Assistant was returning lots of error, mostly related to the version of npm I had. This was due to the fact that my Pis were running on repository that had only older version. Updated everything, resintalled/updated Room-Assistant, no more error.

Some time later, I realized that Room-Assistant seemed to have issue detecting devices presence, mostly mine (Samsung). I've looked at it, I'm not getting any error, it just seems to have trouble detecting my phone. In the past, it usually had more issue with iPhone (probably related to power management) than my Samsung devices. 

I am not sure if it is related to the update (npm, room-assistant) I have done. I also have updated my Samsung to One UI 5.0 and 5.1 (Android 13) recently so I don't know if that could be related?

Anyone noticing issues with their Samsung devices recently?

**Comment by kalto** [comment]
I did some more testing. I seem to have multiple issues. I was able to get one of the Pi to detect my phone by making my phone bluetooth discoverable and moving the phone inches from the Pi. That way, it picked it. Seems like I might have a signal problem or something there as the phone really needed to be close for the Pi to pick it up.

Since then, my phone has been consistently online (±1.5 hour), but always attached to the same Pi even if I moved in the house. Which I found weird.

So I decided to turn off bluetooth in the phone. It's now been off for over 30 minutes and Room-Assistsant still reports the phone as online on the same Pi. I even stopped the service on the Pi detecting my now undetectable phone. The device became "Unavailable" in Home-Assistant.  Rebooted the Pi (with my phone bluetooth still off) and Home Assistant now show my phone to be reachable and attached to the Pi that has just rebooted.

Any idea what might be causing this?

**Comment by embcla** [comment]
I'm having the exact same issue with my S22 and my wife's iPhone 13: the detection rate of both has recently became substantially unreliable.
I have two instances:
 pi in the basement and that is sensibly more consistent
 NAS in the apartment with BT dongle, this has become almost incapable of detecting the devices

There were definitely both HA updates as well as Android ones, wonder which one broke this



Your last few lines, where you say that even having the phone off, it still shows as detected by one of the instances, seem more like an MQTT issue to me than a Bluetooth one, you may try stopping all instances of RA and restarting the MQTT server

**Comment by kalto** [comment]
I have restarted, rebuild, reinstalled, re-whatever-you-can-think-of since then.... to no avail, unfortunately. But on my side, it is only my S22 that seems affected. My wife's iPhone works flawlessly. Mine is being found randomly through the day, but definitely not consistently, making the integration unuseable for my phone (but ok for my wife). Since the project isn't being maintained anymore, I guess I'll have to find an alternative. Still looking for option and, for now, I can't find anything remotely approaching this project.
Still happy and grateful for the time it worked well though. Thanks to mKeRix!
